distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Split_Tokenize.enso

```diff
@@ -0,0 +1,246 @@
+            builders = Vector.new column_count (_-> make_string_builder num_rows)
+
+            output_lengths = 0.up_to num_rows . map i->
+                input_value = input_storage.getItemBoxed i
+                output_values = function input_value
+
+                ## Add `output_values` to builders; if there are more builders
+                   than `output_values`, pad with null.
+                0.up_to builders.length . map i->
+                    builders.at i . appendNoGrow (output_values.get i Nothing)
+
+                output_values.length
+
+            max_output_length = output_lengths.fold 0 .max
+
+            if max_output_length > column_count then
+                problem = Column_Count_Exceeded.Error column_count max_output_length
+                problem_builder.report_other_warning problem
+
+            builders
+
+    # Build Columns.
+    builders.map .seal . map_with_index i->storage->
+        name = input_column.name + "_" + i.to_text
+        Column.from_storage name storage
+
+## PRIVATE
+   Name a set of column vectors to be unique when added to a table.
+rename_new_columns : Table -> Vector Column -> Problem_Builder -> Vector Column
+rename_new_columns table columns problem_builder =
+    unique = Unique_Name_Strategy.new
+    unique.mark_used <| table.columns.map .name
+    new_columns = columns.map column->
+        new_name = unique.make_unique column.name
+        column.rename new_name
+    problem_builder.report_unique_name_strategy unique
+    new_columns
+
+## PRIVATE
+   Replace a single column in a table with new columns.
+replace_column_with_columns : Table -> Column -> Vector Column -> Table
+replace_column_with_columns table old_column new_columns =
+    Table.new ((table.columns.map (c-> if c.name == old_column.name then new_columns else [c])).flatten)
+
+## PRIVATE
+   Return the maximum value of the vector.
+   Throws Empty_Error if the vector is empty.
+maximum : Vector Any -> Vector Any
+maximum vec = if vec.is_empty then Nothing else
+    vec.reduce (a-> b-> a.max b)
+
+## PRIVATE
+   Wrap a function so that it returns Nothing if passed Nothing
+ignore_nothing : (Any -> Any) -> (Any -> Any)
+ignore_nothing function = x-> case x of
+    _ : Nothing -> Nothing
+    _ -> function x
+
+## PRIVATE
+   Asserts that a column exists in the table and is a text column.
+expect_text_column : Table -> Text | Integer -> Problem_Behavior -> Any -> Any
+expect_text_column table column_id ~action =
+    column = table.get column_id
+    case column of
+        Nothing ->
+            Error.throw <| Missing_Input_Columns.Error [column_id]
+        _ : Column ->
+            case Value_Type.is_text column.value_type of
+                False ->
+                    Error.throw <| Invalid_Value_Type.Error Value_Type.Char column.value_type column_id
+                True ->
+                    action
+
+## PRIVATE
+   Repeat a computation n times.
+repeat n ~action = 0.up_to n . map _-> action
```

Comment:
```suggestion
repeat n ~action = 0.up_to n . each _-> action
```

Use `each` to avoid allocating a vector of size `n` which map does (I don't trust our usage analysis to be good enough to remove this allocation, not yet).

I'm not sure if this is needed though, `(0.up_to n).each _-> ...` is kind of an Enso idiom already IMHO and it's more readable than `repeat` - since `repeat` is a function special to this file I first need to look up what it does (although the name is good, so it's not a big deal). I'm OK with keeping it as-is, but my personal and subjective feeling is that using `up_to+each` directly may be as readable as this.

Reply:
I think it's always better to name even small functions like this. At the call site it's easier to understand what's going on. Also, this can be used with an action that returns a value, so this is more general; if we really want a special case that uses each, I would add repeat_each.

Reply:
1. I agree that naming small functions is a good practice. So yeah, I guess you are right to keep it.
2. About this being more general - I don't agree that is a good thing. I looked over the usages and if I'm not mistaken, in no place you keep the return value. So we are allocating a vector and immediately discarding it. I hope in hot paths the JIT would be able to optimize this away, but I'm not certain it will be able to. I see no reason to use `map` over `each` here.

Reply:
I'll rename it to repeat_each. This is the kind of function that I might move to a standard library at some point, so I don't want it to be too specific to what's in this file. But for speed, having one for map and one for each makes sense.